test/mocha/10-api.js

```diff
@@ -607,6 +637,82 @@ describe('bedrock-identity', function() {
           }]
         }, done);
       });
+      it('should update identity to be in group via change set', done => {
+        var userName = '84ae6082-0f48-43ea-bdb1-176b0b6843a2';
+        var groupName = 'ab00d930-f5d3-4887-a4f5-a4f388fc118e';
+        var newIdentity = helpers.createIdentity(userName);
+        var newGroup = helpers.createIdentity(groupName);
+        newGroup.type = ['Identity', 'Group'];
+        newGroup.owner = newIdentity.id;
+        async.auto({
+          insertIdentity: callback => {
+            brIdentity.insert(null, newIdentity, callback);
+          },
+          insertGroup: ['insertIdentity', callback => {
+            brIdentity.insert(null, newGroup, callback);
+          }],
+          update: ['insertGroup', (callback, results) => {
+            const updatedIdentity = newIdentity;
+            const changes = {
+              op: 'add',
+              value: {
+                memberOf: newGroup.id
+              }
+            };
+            brIdentity.update(
+              null, updatedIdentity.id, {changes: changes},
+              callback);
+          }],
+          test: ['update', callback => {
+            database.collections.identity.findOne(
+              {id: database.hash(newIdentity.id)}, (err, results) => {
+                var meta = results.meta;
+                var identity = results.identity;
+                identity.memberOf.should.have.same.members([newGroup.id]);
+                callback();
+              });
+          }]
+        }, done);
+      });
+      it('should allow identity w/ sysResourceRole for a group', done => {
+        var userName = '814b3e02-db14-4fa2-b45b-5d8f977ac087';
+        var groupName = '8314e48c-ff51-427e-aa7b-524cda45b708';
+        var newIdentity = helpers.createIdentity(userName);
+        var newGroup = helpers.createIdentity(groupName);
+        newGroup.type = ['Identity', 'Group'];
+        newGroup.owner = newIdentity.id;
+        newIdentity.memberOf = [newGroup.id];
+        newIdentity.sysResourceRole.push({
+          sysRole: 'bedrock-identity.regular',
```

Comment:
Only collapse when it's the same `delegator` (I record who did the delegating now to allow for revocation in the future) ... we don't have any code that does that collapsing or uniqueness checking yet, but that's what we'd do in the future.